javaparser-core/src/main/java/com/github/javaparser/ast/type/ClassOrInterfaceType.java

```diff
@@ -53,8 +55,14 @@ public ClassOrInterfaceType(final ClassOrInterfaceType scope, final String name)
     }
 
     public ClassOrInterfaceType(final int beginLine, final int beginColumn, final int endLine, final int endColumn,
-            final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
+                                final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
```

Comment:
I also dislike the boolean because that information should actually be inside some class that defines stuff related to the Type Arguments. I'd prefer to encapsulate the `List<Type> typeArgs` inside a class called `TypeArguments` and have the information about the usage of the Diamond Operator inside it. I don't see a problem having a boolean inside that class.

I didn't go that route because it requires more changes and I wanted to get some feedback.

Should I go this route? Any other ideas?